Lib/test/test_array.py

```diff
@@ -1255,6 +1254,19 @@ def test_assignment(self):
         b[slice(2,3)] = ins
         c[2:3:] = ins
 
+    def test_assignment_step_overflow(self):
+        a = array.array(self.typecode, range(10))
+        b = array.array(self.typecode, [100])
+        a[::9999999999] = b
+        self.assertEqual(a, array.array(self.typecode, [100, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
+        a[::-9999999999] = b
+        self.assertEqual(a, array.array(self.typecode, [100, 1, 2, 3, 4, 5, 6, 7, 8, 100]))
+        c = array.array(self.typecode)
+        a[0:0:9999999999] = c
+        self.assertEqual(a, array.array(self.typecode, [100, 1, 2, 3, 4, 5, 6, 7, 8, 100]))
+        a[0:0:-9999999999] = c
+        self.assertEqual(a, array.array(self.typecode, [100, 1, 2, 3, 4, 5, 6, 7, 8, 100]))
+
```

Comment:
Because we can move to new CPython tests in future, we prefer to add RustPython-original tests out of the CPython test code. Would you move this test to `tests/snippets/stdlib_array.py`?